tests/test_main.py

```diff
@@ -0,0 +1,139 @@
+import asyncio
+from datetime import datetime, timezone
+from typing import Any
+from unittest import mock
+from unittest.mock import Mock
+
+import pytest
+from fideslog.sdk.python.event import AnalyticsEvent
+from starlette.requests import Request
+from starlette.responses import Response
+
+from fidesops.analytics import in_docker_container, running_on_local_host
+from fidesops.main import dispatch_log_request
+from fidesops.schemas.analytics import Event
+from fidesops.util.async_util import wait_for
+
+
+class CustomTestException(BaseException):
+    """Mock Non-HTTP Exception"""
+    pass
+
+
+@mock.patch("fidesops.analytics.send_analytics_event")
+@mock.patch("datetime.datetime")
+def test_dispatch_log_request_no_exception(
+        datetime_mock: Mock,
+        send_analytics_request_mock: Mock,
+) -> None:
+    request = Request(
+        scope={
+            "type": "http",
+            "method": 'GET',
+            "path": "/testing",
+            "server": {
+                "host": "www.testsite.com",
+                "port": 3000
+            },
+            "headers": {}
+        },
+    )
+
+    datetime_mock.now.return_value = datetime(2022, 5, 22, tzinfo=timezone.utc)
+
+    f = asyncio.Future()
+    f.set_result(Response(status_code=200))
+
+    def mock_call_next(request: Request) -> Any:
```

Comment:
@pattisdr I'm unsure if you'd have any pointer here, but I'm struggling to mock the return of an async function, while also preserving the `BackgroundTask` that is added to the response here- https://github.com/ethyca/fidesops/pull/622/commits/53f414e53cfa4396a8d27cc68bccd0da1268809d#diff-51868caf44d288309a19f7b59360f148adfa01226f70bf528441cef723439ffcR57

Reply:
have you tried AsyncMocks? https://docs.python.org/3.8/library/unittest.mock.html#unittest.mock.AsyncMock

Reply:
checking this out!